File: leetcode/424. Longest Repeating Character Replacement/sol.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int characterReplacement(string s, int k) {
        vector<int> count(26);
        int maxCount = 0;

        int begin = 0;
        int end = 0;

        int ans = 0;
        while(end < s.size()){
            count[s[end] - 'A']++;
            maxCount = max(maxCount, count[s[end] - 'A']);
            if(end - begin + 1 - maxCount > k){
                count[s[begin] - 'A']--;
                begin++;
            }
            ans = max(ans, end - begin + 1);
            end++;
        }
        return ans;
    }
};

File: 424. Longest Repeating Character Replacement/sol.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        count = {}
        res = 0

        l = 0
        maxf = 0
        for r in range(len(s)):
            count[s[r]] = 1 if s[r] not in count else count[s[r]] + 1
            maxf = max(maxf, count[s[r]])

            if(r - l +1) - maxf > k:
                count[s[l]] -= 1
                l += 1
            
            res = max(res, r - l + 1)
        return res